File: .tmp_py4013/batch_04_sources/2515_shortest_distance_to_target_string_in_a_circular_array.js
// LeetCode 2515 - Shortest Distance to Target String in a Circular Array
// https://leetcode.com/problems/shortest-distance-to-target-string-in-a-circular-array/

/**
 * @param {string[]} words
 * @param {string} target
 * @param {number} startIndex
 * @return {number}
 */
var closestTarget = function(words, target, startIndex) {
    const n = words.length;
    let best = -1;
    for (let i = 0; i < n; i++) {
        if (words[i] === target) {
            let d = i - startIndex;
            if (d < 0) d = -d;
            if (n - d < d) d = n - d;
            if (best < 0 || d < best) best = d;
        }
    }
    return best;
};

File: .tmp_rs4013/rust_check_batch08/2515_shortest_distance_to_target_string_in_a_circular_array.rs
struct Solution;
// LeetCode 2515 - Shortest Distance to Target String in a Circular Array
// https://leetcode.com/problems/shortest-distance-to-target-string-in-a-circular-array/

impl Solution {
    pub fn closest_target(words: Vec<String>, target: String, start_index: i32) -> i32 {
        let n = words.len() as i32;
        let mut best = -1;
        for i in 0..n {
            if words[i as usize] == target {
                let mut d = i - start_index;
                if d < 0 {
                    d = -d;
                }
                if n - d < d {
                    d = n - d;
                }
                if best < 0 || d < best {
                    best = d;
                }
            }
        }
        best
    }
}

fn main() {}
